src/debug-generator/debug-generator.c

```diff
@@ -15,25 +15,46 @@
 #include "proc-cmdline.h"
 #include "recurse-dir.h"
 #include "special.h"
+#include "string-table.h"
 #include "string-util.h"
 #include "strv.h"
 #include "unit-file.h"
 #include "unit-name.h"
 
+typedef enum BootBreakpointType {
+        BOOT_BREAKPOINT_PRE_UDEV,
+        BOOT_BREAKPOINT_PRE_SYSROOT_MOUNT,
+        BOOT_BREAKPOINT_PRE_SWITCH_ROOT,
```

Comment:
Maybe it's useful to add breakpoints for shutdown as well? I figure one after shutdown.target and umount.target but before final.target might be useful as well.

Reply:
In that case we'd have to generalize to "breakpoint" instead of boot breakpoint.

Reply:
Yeah, please name this in a fashion that is open for future extensions.

Reply:
Sure, I'll change that.